src/Sarif/Core/SarifLog.cs

```diff
@@ -68,6 +71,67 @@ public static SarifLog Load(Stream source, bool deferred = false)
             }
         }
 
+        /// <summary>
+        /// Post the SARIF file to an URI that accepts it in a POST method.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown if filePath does not exist.</exception>
+        /// <exception cref="ArgumentNullException">Thrown if postUri/filePath/stream/httpClient is null.</exception>
+        /// <exception cref="HttpRequestException">Throws an exception if the IsSuccessStatusCode property for the HTTP response is false.</exception>
+        /// <param name="postUri"></param>
+        /// <param name="filePath"></param>
+        /// <param name="fileSystem"></param>
+        /// <param name="httpClient"></param>
+        public static async Task Post(string postUri,
+                                      string filePath,
+                                      IFileSystem fileSystem,
+                                      HttpClient httpClient)
+        {
+            if (string.IsNullOrWhiteSpace(filePath))
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            if (!fileSystem.FileExists(filePath))
+            {
+                throw new ArgumentException(nameof(filePath));
+            }
+
+            using Stream fileStream = fileSystem.FileOpenRead(filePath);
+            await Post(postUri, fileStream, httpClient);
+        }
+
+        /// <summary>
+        /// Post the SARIF stream to an URI that accepts it in a POST method.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">Thrown if postUri/stream/httpClient is null.</exception>
+        /// <exception cref="HttpRequestException">Throws an exception if the IsSuccessStatusCode property for the HTTP response is false.</exception>
+        /// <param name="postUri"></param>
+        /// <param name="stream"></param>
+        /// <param name="httpClient"></param>
+        public static async Task Post(string postUri, Stream stream, HttpClient httpClient)
```

Comment:
>string postUri [](http://example.com/codeflow?start=38&length=14)

generally we shouldn't pass strings around as URI arguments, we should use the URI type.